common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -830,26 +830,10 @@ private void lazyExecute0(@Schedule Runnable task) {
 
     private void execute(Runnable task, boolean immediate) {
         boolean inEventLoop = inEventLoop();
-        addTask(task);
         if (!inEventLoop) {
             startThread();
-            if (isShutdown()) {
-                boolean reject = false;
-                try {
-                    if (removeTask(task)) {
-                        reject = true;
-                    }
-                } catch (UnsupportedOperationException e) {
-                    // The task queue does not support removal so the best thing we can do is to just move on and
-                    // hope we will be able to pick-up the task before its completely terminated.
-                    // In worst case we will log on termination.
-                }
-                if (reject) {
-                    reject();
-                }
-            }
         }
-
+        addTask(task);
```

Comment:
> I think this is not correct and it should stay as it was before as otherwise there might be a race.

Thanks for your reply. Regarding the increased competition after the changes you mentioned, I speculate the scenario might be like this: when the executor has already been shutdown, external threads will frequently invoke the startThread method. If this is indeed the concern, I believe we could add a precondition check,
 for example:
 `private void execute(Runnable task, boolean immediate) {
        if(isShutdown()){
            reject();
        }
        boolean inEventLoop = inEventLoop();
        if (!inEventLoop) {
            startThread();
        }
        addTask(task);
        if (!addTaskWakesUp && immediate) {
            wakeup(inEventLoop);
        }
    }`